docs/articles/documentation/guides/basic-guides/assert.md

```diff
@@ -0,0 +1,145 @@
+For example, it can take time for the tested page to send a request to the server for the required data, or an end user's action launches an animation after which the page reaches its final state.
+All these intervals cannot be pre-calculated because they depend on various factors: computer performance,
+network connection speed, etc. In this case, if we perform assertions immediately after the test action, we can get an inconclusive result.
+
+![Asynchronous Functional Testing](../../../images/assertions/asynchronous-testing.png)
+
+An additional timeout is usually added when performing asynchronous functional tests.
+
+![Asynchronous Functional Testing with Extra Waiting](../../../images/assertions/extra-waiting.png)
+
+To stabilize asynchronous functional tests, you need to add a timeout that enables the required changes to be applied successfully.
+Note that adding a timeout can increase the test's duration.
+
+TestCafe uses the smart assertion query mechanism if the assertion receives a [Selector's DOM node state property](select-page-elements.md#define-assertion-actual-value)
+or [client function](obtain-data-from-the-client.md) promise
+as an `actual` value:
+if an assertion did not pass, the test does not fail immediately. The assertion retries to pass multiple times, and
+each time it requests the `actual` property value. The test fails if the assertion could not complete successfully
+within a timeout:
+
+![TestCafe Smart Assertion Query Mechanism](../../../images/assertions/query-mechanism.png)
+
+**Example:**
+
+The following web page is an example:
+
+```html
+<div id="btn"></div>
+<script>
+var btn = document.getElementById('btn');
+
+btn.addEventListener('click', function() {
+    window.setTimeout(function() {
+        btn.innerText = 'Loading...';
+    }, 100);
+});
+</script>
+```
+
+Test code for this page can be as follows.
+
+```js
+test('Button click', async t => {
+    const btn = Selector('#btn');
+
+    await t
+        .click(btn)
+        // A regular assertion fails immediately, but TestCafe retries to run DOM state
+        // assertions many times within the timeout until this assertion passes successfully.
+        // The default timeout is 3000 ms.
+        .expect(btn.textContent).contains('Loading...');
+});
+```
+
+The approach described above allows you to create stable tests with fast runtime and reduces the risk of errors.
+
+You can specify the assertion query timeout in test code by using the [options.timeout](#options) option.
+To set the timeout when launching tests, pass the timeout value to the [runner.run](../../using-testcafe/programming-interface/runner.md#run)
+method if you use an API or specify the [assertion-timeout](../../using-testcafe/command-line-interface.md#--assertion-timeout-ms) option
+if you run TestCafe from the command line.
+
+## Options
+
+* options.timeout
+
+{% include assertions/timeout.md %}
```

Comment:
I guess this is going to result in:

```
## Options

* options.timeout

## options.timeout

...

```

I guess we should **remove headers from the templates** and use them where we paste:

```
## Options

### options.timeout

 {% include assertions/timeout.md %}
```